### ISSUE TITLE
Netatmo wind sensor probably not parsed correctly

### ISSUE THREAD

**Issue by matteofranceschini** [tomvanswam/compass-card#15]
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser

**Release with the issue:**
not relevant
**Last working release (if known):**

**Browser and Operating System:**
safari,chrome, homeassistant app, MacOs

**Description of problem:**

I have a Netatmo wind sensor, but the default HA integration provides the information with both text and degrees, for example like `W (240°)`, as shown in picture.

I think that there's a problem for the compass-card to parse this kind of information as, I assume, it expects just one of the two information and not both.

As you can see, it does not crash or anything, but obviously the arrow does not point in the correct direction!

<img width="365" alt="image" src="https://user-images.githubusercontent.com/3541957/90120544-2625df00-dd5b-11ea-9f5c-811edd93e425.png">
 For reference, this is the code that generates this card:

```
cards:
  - detail: 2
    entity: sensor.netatmo_jori_palace_anemogio_gust_strength
    graph: line
    name: Raffiche
    type: sensor
  - direction_offset: 0
    entity: sensor.netatmo_jori_palace_anemogio_angle
    name: Vento
    type: 'custom:compass-card'
    secondary_entity: sensor.netatmo_jori_palace_anemogio_wind_strength
type: horizontal-stack

```
I'm not fluent in javascript nor python, but i think that the problem could just be that the code just checks if the value given is NaN and does not search for a number in the string, to be parsed.

**Javascript errors shown in the web inspector (if applicable):**

```

```

**Additional information:**
If needed, I can do some testing.

**Comment by tomvanswam** [comment]
You're correct, only degrees or abbreviation is possible at the moment (as stated in the README).

I can parse for the numbers and only use those, so I'll change this to an enhancement.

**Comment by matteofranceschini** [comment]
Thank you! As a C programmer that triple equal sign gave me a heart attack, but it seems I‘ve just discovered one of the wonders of javascript :)

I’ll test your edit on monday, when I have a little time !

Ps: I love the design of your compass card, it’s a strange thing that we don’t have it natively in Homeassistant. Thank you for this magnificent work!

**Comment by tomvanswam** [comment]
I think I got it, a input_text entity helps a lot with testing this as you can set it to whatever you want.

One note though, currently all characters which are not numbersor dots are removed from the entity state. If the state is something like `North 342 degrees @ 15kn` it will be parsed to `342615` in which all multiples of 360 are removed so the resulting direction will become  `255`.

My RegEx-fu is not strong enough yet (or my google skills fail me) at only getting the first number sequence in the string and only use that.

Workaround in all not supported cases would be
- create a [template sensor](https://www.home-assistant.io/integrations/template/) 
- [parse](https://www.home-assistant.io/docs/configuration/templating/) your entity in a correct way so you get 0-360 degrees into the value_template of said sensor
- use the template sensor in the compass-card

**Comment by matteofranceschini** [comment]
If you can use a regex, you could try with something like this:

`\b\d?\d?\d\b`

This should get you just the first time you find _at least 1 number, maximum 3, and no other number, if any, afterwards._

I'm not a regex expert, but I've tried it and it seems to get the result we want :)

**Comment by tomvanswam** [comment]
I updated the regex a bit, your propsal inspired me to have a second look.

First I replace all the whitespaces now, so any negative sign in front of the number would have a space removed if it was in between.  
Second I fixed the negative directions (was not working for the parsed string).
And third I search for all consecutive numbers + a dot and max 1 decimal (also consecutive). Should be good enough I think.

Note, I'll put this in the next version (0.2.1, or 0.3.0, depending on the other changes in the next version)
